src/Agent.Listener/Agent.cs

```diff
@@ -216,6 +216,13 @@ public async Task<int> ExecuteCommand(CommandSettings command)
                 Trace.Info($"Set agent startup type - {startType}");
                 HostContext.StartupType = startType;
 
+                bool debugModeEnabled = command.GetDebugMode();
+                settings.DebugMode = debugModeEnabled;
+                store.SaveSettings(settings);
+                if (debugModeEnabled)
+                {
+                    Trace.Warning("Agent is running in debug mode, don't use it in production");
```

Comment:
[not blocking for this PR] Does this bubble to the timeline?  We might want to write the warning there

Reply:
Good point! I'll improve this when I raise a PR for debugging powershell tasks.